pkgs/development/compilers/zig/0.10.nix

```diff
@@ -62,27 +60,36 @@ stdenv.mkDerivation rec {
     "-DZIG_TARGET_MCPU=baseline"
   ];
 
+  preBuild = ''
+    export HOME=$TMPDIR;
+  '';
```

Comment:
This is here in order to prevent `~/.cache/zig` from being written to. Here is the upstream logic for choosing that directory:

https://github.com/ziglang/zig/blob/0.10.1/src/introspect.zig#L101-L132

Do you have any suggested improvement that would allow us to clean this up and omit `preBuild`?

Reply:
Well, if I am undertstanding your point, the Zig compiler writes and reads cache files, even as part of its own bootstrapping.

Given this, the options I can envise are
- disabling cache (via env vars or cmdline params)
- setting cache directory (via env vars or cmdline params)

If I understood the code, `ZIG_GLOBAL_CACHE_DIR` is the variable created in order to hold this cache. Therefore, in a first sight, I would use something like

```nix
env.ZIG_GLOBAL_CACHE_DIR = "$NIX_BUILD_TOP/zig-cache";
```

or something similar.
However I do not remember if this is allowed.

Reply:
I think the code to set `HOME` (or even better would be `XDG_CACHE_HOME`) is fine. My offer is to make an upstream change that simplifies the nix package. However, I do not see any possible improvement. Perhaps it is fine as-is, and it is expected for nix to override the cache directory as part of the package.

Reply:
The problem is that, as I have said, Zig itself needs a cache directory. 
Then this cache needs to be set somehow.
In light of this, the best you can do is already done: an env var that holds this cache (and a failsafe default).

I  have changed this code accordingly.

Reply:
> However I do not remember if this is allowed.

https://github.com/NixOS/nixpkgs/issues/189691

Reply:
I will use TMPDIR then